crates/cubecl-runtime/src/tune/tuner.rs

```diff
@@ -16,6 +19,10 @@ use crate::tune::{AutotuneOperation, AutotuneOperationSet, TuneBenchmark, TuneCa
 
 use super::AutotuneKey;
 
+/// An error that occured during benchmarking. If other benches succeeded, ignore this bench and
+/// continue gracefully. If all benches fail, panic.
+type BenchError = ManuallyDrop<Box<dyn Any + Send>>;
```

Comment:
Is there a reason to use that type for the error? Can we create a custom error type that we map instead?

Reply:
Unfortunately not. The way panics work in rust means that the error type returned by `catch_unwind` is completely opaque and afaik doesn't even contain the panic message, instead containing the unwind logic and stack. That's why it has to be `ManuallyDrop`; dropping it can actually trigger the unwind and cause all sorts of havoc in the process. That's why instead of providing a custom error message, we just have to pass it to `resume_unwind`, it's the only function that knows how to do anything with this opaque blob.

Reply:
I would actually add that explaination in the code as a comment, that's very clear thanks!

Reply:
Done 👍